oeps/oep-0011.rst

```diff
@@ -0,0 +1,299 @@
+================================
+OEP-11: Front End Best Practices
+================================
+
++---------------+-------------------------------------------+
+| OEP           | :doc:`OEP-11 </oeps/oep-0011>`            |
++---------------+-------------------------------------------+
+| Title         | Front End Best Practices                  |
++---------------+-------------------------------------------+
+| Last Modified | 2016-10-19                                |
++---------------+-------------------------------------------+
+| Author        | Andy Armstrong <andya@edx.org>            |
++---------------+-------------------------------------------+
+| Arbiter       | TBD                                       |
++---------------+-------------------------------------------+
+| Status        | Draft                                     |
++---------------+-------------------------------------------+
+| Type          | Best Practice                             |
++---------------+-------------------------------------------+
+| Created       | 2016-10-19                                |
++---------------+-------------------------------------------+
+
+Abstract
+========
+
+This document describes the best practices for developing front end applications
+and components for Open edX. It covers the techniques and approaches that
+are currently recommended, as well as the suggested technologies and libraries
+that should be used.
+
+
+Specification
+=============
+
+Best Practices
+~~~~~~~~~~~~~~
+
+1. **Front ends should render themselves server-side as much as possible**
+
+   Rationale: Server-side rendering gives the content to the user
+   immediately without requiring additional round-trips. This can make
+   for large performance improvements on slow networks or mobile devices.
+
+   If a dynamic client-side experience is needed, then the server-side
```

Comment:
I think we should describe the different types of "front-ends". We have many server-side rendering Django apps and then we have single-page apps. I think this document would still be useful for server-side apps, because we also talk about things like the pattern library, accessibility, internationalization, and etc. which are all still relevant to fully server-side apps.